### PR TITLE
We like Taylor Series: add multi-dim expansions

### PR DESCRIPTION
**Pull request by latot**
Hi again, now we can have the taylor series in multivariable (with all vars you want).

Fixes https://github.com/cbm755/octsympy/issues/31

Cya.

### CODE REVIEW

File: inst/@sym/taylor.m [hunk 1]
```diff
@@ -266,9 +278,3 @@
 %! g = x^2 + 2*x + 3;
 %! h = taylor (g, x, 4, 'order', 1);
 %! assert (isequal (h, 27))
-
-%!error<Only 2d expansions supported.>
```
Comment (cbm755) [inline inst/@sym/taylor.m]: can make it into a test instead of removing it?

**Comment by cbm755** [comment]
Can you add a test or two to verify it works?

**Comment by latot** [comment]
okidokis, more tests!, the results match with wolframalpha (but sympy its very slow compared with it)

**Comment by latot** [comment]
only to be sure i compare the examples of SMT with this and are equals (the only difference is here are simplified).

**Comment by latot** [comment]
i feel i'm giving to you a lot of work with all this pr...

File: inst/@sym/taylor.m [hunk 2]
```diff
@@ -89,6 +89,9 @@
     % taylor(f, [x,y], [a,b], 'param', ...)
     x = varargin{1};
     a = varargin{2};
+    if length(a) ~= length(x) && length(a) == 1
```
Comment (cbm755) [inline inst/@sym/taylor.m]: this happens if `length(x) == `1 and `length(a) ~= 1`...?

Should it be: `if (length(a) == 1) && (length(x) > 1)`?

File: inst/@sym/taylor.m [hunk 3]
```diff
@@ -117,28 +120,40 @@
...
 
+  if isa(n, 'sym')
+    n = double(n);
+  end
+
   if (isfloat(n))
     n = int32(n);
   end
 
+  assert( isequal( length(x), length(a)), 'The length of the expansion point must be the same as the input variables.')
```
Comment (cbm755) [inline inst/@sym/taylor.m]: this should have a BIST like `%!error <length> taylor(...)`

File: inst/private/python_header.py [hunk 4]
```diff
@@ -44,6 +44,7 @@ def echo_exception_stdout(mystr):
     import codecs
     import xml.etree.ElementTree as ET
     from distutils.version import LooseVersion
+    import itertools
```
Comment (cbm755) [inline inst/private/python_header.py]: I think this will have to go in `python_ipc_pytave.m` as well.

File: inst/@sym/taylor.m (hunk 2 above)
Comment (latot) [inline inst/@sym/taylor.m]: mm half half, this should be an error:
```taylor(0, [x, y], [1, 2, 3])```
sadly don't are equals.

**Comment by latot** [comment]
This code is more slower compared with the actual 2D, because the actual is restricted to this:
```
     2                                   2          
    ∂                                  ∂            
  ─────(f(x, y))    ==     ─────(f(x, y))
  ∂y ∂x                            ∂x ∂y  
```
but this only is true when we have continuous diffs, and in some discontinuous diff, this code don't are restricted with this.
Actually are ways to have better performance to this, i put a ```FIXME``` for the moment, i think for at least have the base for multivar should be fine.

Code Updated!

File: inst/@sym/taylor.m [hunk 5]
```diff
@@ -117,28 +120,40 @@
     end
   end
 
+  if isa(n, 'sym')
+    n = double(n);
+  end
+
```
Comment (cbm755) [inline inst/@sym/taylor.m]: Why is this important?

**Comment by cbm755** [comment]
@genuinelucifer does this look ok?  In particular, can you sign off on the pytave change?

File: inst/@sym/taylor.m (hunk 5 above)
Comment (latot) [inline inst/@sym/taylor.m]: n can't have symbols (x, y, z.....), maybe use findsymbols? or remove it and lets python handle an error?

Comment (latot) [inline inst/@sym/taylor.m]: mm, lets left this with the python error, this can work in the future:
```
syms x n
taylor(x, 'order', n)
```

**Comment by latot** [comment]
Okidokis!

**Comment by genuinelucifer** [comment]
+1 from me. The changes look good (after the sym to double removal)...

**Comment by latot** [comment]
Yay! Thx.

**Comment by bernamdc** [comment]
I'm not sure how github works. But I did not find a working multivariate Taylor expansion so I developed one. You can find it in my answer in:

[https://stackoverflow.com/questions/22857162/multivariate-taylor-approximation-in-sympy/63850672#63850672]

Hope it helps for something!

**Pull request merged**